tests/test_main.py

```diff
@@ -666,6 +666,28 @@ def current_lessequal_500(cls, values):
     ]
 
 
+def test_root_validator_many_values_change():
+    """It should run root_validator on assignment and update ALL concerned fields"""
+
+    class Rectangle(BaseModel):
+        width: float
+        height: float
+        area: float = None
+
+        class Config:
+            validate_assignment = True
+
+        @root_validator
+        def set_ts_now(cls, values):
```

Comment:
Wrong validator function name from the issue snippet, my bad
```suggestion
        def set_area(cls, values):
```